content/en/blog/_posts/2021-01-28-PSP-deprecation.md

```diff
@@ -0,0 +1,49 @@
+
+To quote the above video: "Various limitations and structural problems have prevented the PSP API from GA." Having objects in Kubernetes never reaching GA (“permabeta”) is a problem for administrators and organizations, as a non-GA feature is not considered to be ready for widespread use. Given the structural problems with the PSPs as they exist today, there is no real path for them to become GA. Per the deprecation issue, “the future of restricting pod security settings does not lie in PSP because compatibility restrictions will prevent the kinds of changes that are required.” As such, PSPs will be deprecated in Kubernetes 1.21, though they will not be fully removed until a future version.
+
+
+## What will replace them? 
+
+Securing your pods requires understanding how the pods are intended to run, and what types of behaviors or activities you seek to prevent. While Pod Security Policies have been one way to enforce security configurations, other tools in the cloud native ecosystem have security enforcement capabilities. The [Pod Security Standards](/docs/concepts/security/pod-security-standards/) page in the Kubernetes documentation aims “to detail recommended Pod security profiles, decoupled from any specific instantiation.”  This page is a useful reference for common pod security configurations, which can be implemented by native or external pod security enforcement tools.
+
+While the scope of PSPs is beyond anything that will replace it, there are [a few tools in the ecosystem](https://landscape.cncf.io/card-mode?category=security-compliance&grouping=category) that can help enforce security policies through the management of Admission Controller Webhooks. We encourage you to assess the right tool for your needs. Pod Security Policies are one type of Admission Controller Webhook in Kubernetes, but there are a variety of available options. The blog post, [“A Guide to Kubernetes Admission Controllers”](/blog/2019/03/21/a-guide-to-kubernetes-admission-controllers/) from March 2019 gives a great overview of what admission controller webhooks can do.
+
+While the exact capabilities and use of policy enforcement tools will vary, the community is engaged in working toward better and better solutions for enforcing security policies across your Kubernetes pods.
+
+
+## Wrapping it up
+
+Deprecation notices draw our attention to the wide range of work going on in the Kubernetes ecosystem that will change the way we operate Kubernetes clusters in the future. This work is taken quite seriously and goes through a detailed exploration process by related SIG leads to understand the implications.
+
+The deprecation of PSP will begin with Kubernetes version 1.21, with its removal in a future version. As we approach that milestone, the Kubernetes community will be working to document viable alternatives.
```

Comment:
It's slotted to be removed in 1.25 but, that's far enough out for it to slip or move. We didn't necessarily want to lock the community into a hard and fast date/release.

Reply:
+1 for putting a concrete end-release instead of a vague "future version". The removal release is explicitly 1.25, and is surfaced in release notes and in warnings returned by the API server.

Reply:
A new commit has been pushed to address this.